src/server/lesson-controller.js

```diff
@@ -37,6 +37,10 @@ export default class LessonController {
     const { lessonId } = req.params;
     const routeWildcardValue = urls.removeLeadingSlash(req.params['0']);
 
+    if (!user) {
+      return res.redirect(urls.getLoginUrl(req.path));
```

Comment:
We usually do not manually redirect to the login page because the whole logic for redirecting and navigating back to the original page is already implemented in the global error handler. If authentication is required we should throw a `new Unauthorized()` (with "Unauthorized" actually meaning status 401).

Having that said, this check has to come further down (to around line 58) as it is only required to be authenticated when navigating to a private room, but this clause here prevents anonymous users from accessing any room, including public ones (as far as I can see).

Reply:
Then let's revisit our access requirements for rooms and lessons because:
- both now use the permissions middleware, which means none can be accessed by anonymous users
- controllers for both pages reach the `user._id` part of the code if the session is deleted (which indeed should not happen having the permissions middleware)
- I was under the possibly false impression that rooms (and thus lessons as well), even if public, they would require authentication (reason for which we had permissions requirements on rooms already)